samples/bluetooth/central_path_loss_monitoring/src/main.c

```diff
@@ -0,0 +1,235 @@
+    return;
+  }
+  printk("Connection to %s\n", addr);
+}
+
+/* Initialize a scan callback structure */
+BT_SCAN_CB_INIT(scan_cb, scan_filter_match, NULL, NULL, NULL);
+
+/* This function runs after a successful or failed connection attempt */
+static void connected(struct bt_conn *conn, uint8_t conn_err) {
+  char addr[BT_ADDR_LE_STR_LEN];
+  bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+  if (conn_err) {
+    printk("Failed to connect to %s (err 0x%02x)\n", addr, conn_err);
+    bt_conn_unref(default_conn);
+    default_conn = NULL;
+    bt_scan_start(BT_SCAN_TYPE_SCAN_ACTIVE);
+    is_connected = false;
+    return;
+  }
+  printk("Connected to %s\n", addr);
+  is_connected = true;
+
+  /* Define path loss monitoring parameters */
+  struct bt_conn_le_path_loss_reporting_param pl_param = {
+      .high_threshold = HIGH_PATH_LOSS_BORDER,
+      .high_hysteresis = PATH_LOSS_BUFFER_ZONE,
+      .low_threshold = LOW_PATH_LOSS_BORDER,
+      .low_hysteresis = PATH_LOSS_BUFFER_ZONE,
+      .min_time_spent = NUM_ITERATIONS};
+
+  /* Set the parameters */
+  int err = bt_conn_le_set_path_loss_mon_param(conn, &pl_param);
+  if (err) {
+    printk("Failed to set path loss monitoring parameters (er: %d)\n", err);
+    return;
+  }
+
+  /* Enable path loss monitoring */
+  err = bt_conn_le_set_path_loss_mon_enable(conn, true);
+  if (err) {
+    printk("Failed to enable path loss monitoring (err: %d)\n", err);
+    return;
+  }
+  printk("Path loss monitoring enabled\n");
+}
+
+/* Function when disconnection occurs */
+static void disconnected(struct bt_conn *conn, uint8_t reason) {
+  char addr[BT_ADDR_LE_STR_LEN];
+  bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+  printk("Disconnected from %s (reason: 0x%02x)\n", addr, reason);
+
+  if (default_conn == conn) {
+    bt_conn_unref(default_conn);
+    default_conn = NULL;
+    bt_scan_start(BT_SCAN_TYPE_SCAN_ACTIVE);
+  }
+  is_connected = false;
+  polling_started = false;
+}
+
+/* Initializes the scanning process */
+static void scan_init(void) {
+  /* Scan parameters with fast scanning intervals and windows */
+  struct bt_le_scan_param scan_param = {
+      .type = BT_LE_SCAN_TYPE_ACTIVE,
+      .interval = BT_GAP_SCAN_FAST_INTERVAL,
+      .window = BT_GAP_SCAN_FAST_WINDOW,
+  };
+
+  /* Struct to initialize the scan module */
+  struct bt_scan_init_param scan_init = {
+      .connect_if_match =
+          DISABLED,              /* Disable automatic connection. Instead handle
+                                  * connection manually through scan_filter_match()
+                                  */
+      .scan_param = &scan_param, /* Links to a bt_le_scan_param structure,
+                                    defined earlier */
+  };
+
+  bt_scan_init(&scan_init);
+  bt_scan_cb_register(&scan_cb);
+
+  int err = bt_scan_filter_add(BT_SCAN_FILTER_TYPE_UUID, &bt_uuid.uuid);
+  if (err) {
+    printk("Failed to add UUID filter (err %d)\n", err);
```

Comment:
Could we remove all printks and use log_debug/log_error instead? then its easier to remove all prints by just changing the debug level